### ISSUE TITLE
[Request] Pinned items inside when they out of their parent constraints (X axis)

### ISSUE THREAD

**Issue by burekas7** [AhmedLSayed9/dropdown_button2#86]
Hi,

Can you add an automatic behavior when the DropDownItems are outside the range on X axis to align it inside (As you do when it outside for y axis), without need to set the offset option.

![image](https://user-images.githubusercontent.com/13424257/194711101-e766800a-aadc-4307-b201-4f5d0211cc31.png)![image](https://user-images.githubusercontent.com/13424257/194711218-5bdf11c2-bbfc-4568-a80b-03ee3332c015.png)

```
return Scaffold(
      appBar: AppBar(
        title: Text("Title"),
      ),
      body: SafeArea(
          child: Stack(
        children: [
          Positioned(
            bottom: 70,
            left: 30,
            child: DropdownButton2(
              underline: Container(),
              dropdownDecoration: BoxDecoration(
                borderRadius: BorderRadius.circular(14),
                color: Colors.redAccent,
              ),
              dropdownWidth: 150,
              customButton: Container(
                padding: const EdgeInsets.all(8),
                decoration: BoxDecoration(
                  borderRadius: BorderRadius.circular(14),
                  border: Border.all(
                    color: Colors.black26,
                  ),
                  color: Colors.redAccent,
                ),
                child: const Icon(
                  Icons.sort,
                  color:
                      Colors.black, //Theme.of(context).primaryIconTheme.color,
                ),
              ),
              items: const [
                DropdownMenuItem(value: 1, child: Text("Item 1")),
                DropdownMenuItem(value: 2, child: Text("Item 2")),
                DropdownMenuItem(value: 3, child: Text("Item 3")),
              ],
              onChanged: (value) {},
            ),
          )
        ],
      )
    );
```

**Comment by AhmedLSayed9** [comment]
This behavior is already there.
Dropdown menu doesn't render outside the range of the screen even if you specified `Offset` that should render it outside.
Your sample code doesn't render outside for me as your screenshot. Can you try it again?

**Comment by burekas7** [comment]
Strange.
I tested it on Android (Real device + Emulator), on both it acts the same behavior as the image I posted above.
On x axis it go out of the screen when the drop items are out of screen bounds. Not like it acts on y axis.

(Also tested on 1.8.3)

**Comment by AhmedLSayed9** [comment]
https://user-images.githubusercontent.com/70890146/194728110-05730d73-3f8b-45d7-96da-537ea8108eb0.mov

Also, I've tested it on iOS. You might test a different code than the above sample.

**Comment by burekas7** [comment]
I understand why.
I forgot to mention that my locale set as RTL ( Locale('he', 'IL') ),
This is why the direction is the opposite on my image and my testing.

**Comment by AhmedLSayed9** [comment]
Alright! Provide another sample that can produce the issue.

**Comment by burekas7** [comment]
The code is the same but you need to set locale for MaterialApp, but it requires some additional files for that etc.. so it's not just a sample.

It seems that for LTR it does work when the button is positioned on the right side of the screen.
Do you support both options of LTR and RTL?

**Comment by AhmedLSayed9** [comment]
Finally, I was able to produce the issue.
The package is supporting directionality so I'll work to fix this. Thanks!

**Comment by burekas7** [comment]
Thanks, it is fixed and working with version 1.8.4
